### ISSUE TITLE
PyEmma not preserving order of input files with duplicates?

### ISSUE THREAD

**Issue by jhprinz** [markovmodel/PyEMMA#1066]
This is more a quick question on whether the returned list of trajectory lengths and discrete trajectories are the same as the input trajectories. Example

```py
files = [ 'tLen10.dcd', 'tLen20.dcd', 'tLen30.dcd' ]
inp = pyemma.coordinates.source(files, feat)
print inp.trajectory_lengths()
>>> [10, 20, 30]  # or something else
```

If not, is there a way to enforce this? Actually, I had the order changed, but accidentally had the same file twice. Also, what is the behaviour on duplicates. The number of listed trajectories suggested that the duplictes are listed multiple times.

**Comment by jhprinz** [comment]
After more analysis this really seems only to happen when there are duplicate filenames present. In that case let me rephrase the question. That is the behaviour when having duplicate filenames!

**Comment by jhprinz** [comment]
Seems I miscounted the returned trajectories from pyemma. So apparently it uses only each trajectory once and this messes up the order. A long as this is preserved when all filenames are unique my problem is solved.

**Comment by marscher** [comment]
Hi,

duplicated files are kicked out of the list. Please correct me, if I'm 
wrong, bug it does not really make sense to feed in the same files a 
couple of times.

**Comment by jhprinz** [comment]
Well, could be when reweighting, but that is merely a theoretical idea. Usually no, you are right. It just seemed that I ran two simulations and used the same name and then suddenly all went weird.

Thanks for the response!

**Comment by marscher** [comment]
probably I should use a more sophisticated dupe detection. Currently it 
is only based on the file names, but it could rely on the TrajInfo, 
which would then also check size, mtime and fingerprint.

**Comment by gph82** [comment]
What about printing a warning?

**Comment by marscher** [comment]
This warning is being logged.
